pandas/tests/io/formats/style/test_style.py

```diff
@@ -746,6 +746,22 @@ def color_negative_red(val):
         df.loc[pct_subset]
         df.style.applymap(color_negative_red, subset=pct_subset)
 
+    @pytest.mark.parametrize(
+        "stylefunc", ["background_gradient", "bar", "text_gradient"]
+    )
+    def test_subset_for_boolean_cols(self, stylefunc):
+        # GH47838
+        df = DataFrame(
+            [
+                [1, 2],
+                [3, 4],
+            ],
+            columns=[False, True],
+        )
+        styled = getattr(df.style, stylefunc)()
+        styled._compute()
+        assert set(styled.ctx) == {(0, 0), (0, 1), (1, 0), (1, 1)}
```

Comment:
Can the entire `styled.ctx` be compared instead of the `set` of it? Since sets do no guarantee order I'm weary this may break in the future.

Reply:
consider `styled.ctx.keys() <= {(0, 0), (0, 1), (1, 0), (1, 1)}`

Reply:
actually doesn't matter since set elements are compared `{"a", "b"} == {"b", "a"}  # TRUE`